docs/Using-Primary-Outputs-for-Post-Actions.md

```diff
@@ -0,0 +1,408 @@
+  "identity": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "shortName": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "sourceName": "MyTestProject",
+  "sources": [
+    {
+      "source": "./Src/Custom/Path/",
+      "target": "./Target/Output/"
+    }
+  ],
+  "primaryOutputs": [
+    {
+      "path": "Target/Output/MyTestProject.csproj"        
+    }
+  ],
+  "postActions": [
+    {
+      "description": "Restore NuGet packages required by this project.",
+      "manualInstructions": [
+        { "text": "Run 'dotnet restore'" }
+      ],
+      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
+      "continueOnError": true,
+    }
+  ]
+}
+```
+
+## Example 2 - reference source files in post action arguments
+The following example restores individual files specified in `files` property. Defining them in primary outputs is optional.
+template files:
+- `./Src/Custom/Path/MyTestProject.csproj`
+- `./Src/Custom/Path/Class.cs`
+`template.json` definition
+```
+{
+  "author": "Test Asset",
+  "classifications": [ "Test Asset" ],
+  "name": "TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "generatorVersions": "[1.0.0.0-*)",
+  "groupIdentity": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "precedence": "100",
+  "identity": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "shortName": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "sourceName": "MyTestProject",
+  "sources": [
+    {
+      "source": "./Src/Custom/Path/",
+      "target": "./Target/Output/"
+    }
+  ],
+
+  "primaryOutputs": [
+    {
+      "path": "Target/Output/MyTestProject.csproj"           
+    }
+  ],
+
+  "postActions": [
+    {
+      "description": "Restore NuGet packages required by this project.",
+      "manualInstructions": [
+        { "text": "Run 'dotnet restore'" }
+      ],
+      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
+      "continueOnError": true,
+      "args": {
+        "files": [ "Src/Custom/Path/MyTestProject.csproj" ]  //here the source location should be used as the post action supports this notation
+      }
+    }
+  ]
+}
+```
+Supported for:
+- restore NuGet packages (`dotnet CLI`)
+- add reference to a project file (`dotnet CLI`)
+
+## Example 3 - using primary output indexes in post action arguments
+
+Opens the file from primary outputs defined by index.
+`template.json` definition
+```
+"primaryOutputs": [
+  { "path": "Company.ClassLibrary1.csproj" },
+  { "path": "Class1.cs" }
+],
+"postActions": [
+  {
+    "description": "Opens Class1.cs in the editor",
+    "manualInstructions": [ ],
+    "actionId": "84C0DA21-51C8-4541-9940-6CA19AF04EE6",
+    "args": {
+      "files": "1"
+    },
+    "continueOnError": true
+  }
+]
+```
+
+Supported for:
+- add projects to a solution file (`dotnet CLI`)
+- add reference to a project file (Visual Studio)
+- open a file in the editor (Visual Studio)
```

Comment:
Is this post action supported by these hosts, or is "using primary output indexes in post action arguments" supported by these hosts?

I understand the latter. But if that's not true, than a bit more clarification here could help.

Reply:
Also, should it be "supported by" or "supported in" instead of "supported for"? I'm not really sure.